tests/test_schema.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of ci_hsc_gen3.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+import yaml
+
+from lsst.daf.butler import Butler
+import lsst.utils.tests
+
+from lsst.utils import getPackageDir
+
+
+class TestSchemaMatch(lsst.utils.tests.TestCase):
+    """Check the schema of the parquet outputs match the DDL in sdm_schemas"""
+
+    def setUp(self):
+        self.butler = Butler(os.path.join(getPackageDir("ci_hsc_gen3"), "DATA"), writeable=False,
+                             collections=["HSC/runs/ci_hsc"])
+        schemaFile = os.path.join(getPackageDir("sdm_schemas"), 'yml', 'hsc.yaml')
+        with open(schemaFile, "r") as f:
+            self.schema = yaml.safe_load(f)['tables']
+
+    def _validateSchema(self, dataset, dataId, tableName):
+        """Check the schema of the parquet dataset match that in the DDL.
+        Only the column names are checked currently.
+        """
+        sdmSchema = [table for table in self.schema if table['name'] == tableName]
+        self.assertEqual(len(sdmSchema), 1)
+        expectedColumnNames = set(column['name'] for column in sdmSchema[0]['columns'])
+
+        df = self.butler.get(dataset, dataId)
```

Comment:
`outputColumnNames = butler.get(dataset + ".columns", dataId)` will give you the columns directly

Reply:
Thanks for pointing it out. It seems a useful feature. 

However, the columns it returns contain one extra, non-SDM, column (`id`). I could just ignore that, but also feel this is tied to daf_butler `ParquetFormatter`'s implementations.  Right now it uses the `metadata.schema` from `pyarrow.parquet` and it's just how it is. I don't feel like I can declare that as a bug either.  But if we want to remove this extra column maybe we should do that in daf_butler instead?  

Given that, I'm inclined not to use the shorthand here until we sort it out on the daf_butler side.